### ISSUE TITLE
Exception with version 4

### ISSUE THREAD

**Issue by marbat87** [mikepenz/Android-Iconics#445]
## About this issue

Hi,
I enabled Iconic with:
`super.attachBaseContext(newBase.wrapByIconics())`

I got an exception on activity start with the latest 4.0.0-b1 version.
The exception is the following:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{it.cammino.risuscito/it.cammino.risuscito.MainActivity}: android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class LinearLayout
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
     Caused by: android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class LinearLayout
     Caused by: android.view.InflateException: Binary XML file line #24: Error inflating class LinearLayout
     Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter parent
        at com.mikepenz.iconics.context.IconicsLayoutInflater.onCreateView(IconicsLayoutInflater.kt)
        at androidx.core.view.LayoutInflaterCompat$Factory2Wrapper.onCreateView(LayoutInflaterCompat.java:55)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:769)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:495)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
        at com.android.internal.policy.DecorView.onResourcesLoaded(DecorView.java:1813)
        at com.android.internal.policy.PhoneWindow.generateLayout(PhoneWindow.java:2554)
        at com.android.internal.policy.PhoneWindow.installDecor(PhoneWindow.java:2627)
        at com.android.internal.policy.PhoneWindow.getDecorView(PhoneWindow.java:2047)
        at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:575)
        at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:518)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:466)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
        at it.cammino.risuscito.MainActivity.onCreate(MainActivity.kt:116)
        at android.app.Activity.performCreate(Activity.java:6679)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

It seems that both IconicsLayoutInflater and IconicsLayoutInflater2 have an incorrect override, in particular, this one:
```
class IconicsLayoutInflater(
    private val appCompatDelegate: AppCompatDelegate
) : LayoutInflaterFactory {

    override fun onCreateView(
        parent: View,
        name: String,
        context: Context,
        attrs: AttributeSet
    ): View? {
        val result = appCompatDelegate.createView(parent, name, context, attrs)
        return IconicsFactory.onViewCreated(result, context, attrs)
    }
```

parent variable is nullable so the correct definition shoule be:
```
override fun onCreateView(
            parent: View?,
            name: String,
            context: Context,
            attrs: AttributeSet
    ): View? {
        val result = appCompatDelegate.createView(parent, name, context, attrs)
        return IconicsFactory.onViewCreated(result, context, attrs)
    }
```

## Details
- [4.0.0-b1] Used library version

**Comment by mikepenz** [comment]
@marbat87 thank you very much for the report

**Comment by zTrap** [comment]
Resolved via #446

**Comment by mikepenz** [comment]
v4.0.0-rc01 released which should contain the fix
